t/032_cancel_consume_ack_exception.t

```diff
@@ -0,0 +1,40 @@
+use Test::More tests => 17;
+use strict;
+use warnings;
+
+use FindBin qw/$Bin/;
+use lib "$Bin/lib";
+use NAR::Helper;
+
+my $helper = NAR::Helper->new;
+
+ok $helper->connect, "connected";
+ok $helper->channel_open, "channel_open";
+ok $helper->exchange_declare, "default exchange declare";
+ok $helper->queue_declare, "queue declare";
+ok $helper->queue_bind, "queue bind";
+ok $helper->drain, "drain queue";
+
+#we don't need to publish or anything, as the segfault happens either way
+my $rv = $helper->recv(-1);
+$helper->ack( 1231211 );
+
+#cancel will fail because the above ack caused an error
+# this used to cause a segfault. but now it shouldn't
+is $helper->cancel, undef, 'cancel fails';
```

Comment:
This test actually forces `$helper->{consumer_tag}` to be used as the consumer tag for the `basic_cancel()` operation since `NAR::Helper`'s cancel is coded thusly:

```perl
sub cancel {
    my ( $self, $tag ) = @_;

    if ( !defined $tag ) {
        $tag = $self->{consumer_tag};
    }
    $self->_ok( sub {
        $self->mq->cancel( $self->{channel}, $tag );
    } );
}
```

We should either remove that conditional, or - my preference - send a fake consumer tag in to catch that extra conditional.